### PR TITLE
Improvements on `OverrideDuplicateMove`

### PR DESCRIPTION
**Pull request by brunoperdigao**
This PR aims to improve IFC object duplication, by implementing a solution for #3770, #2187, #3789.

### CODE REVIEW

File: src/blenderbim/blenderbim/bim/module/geometry/operator.py [hunk 1]
```diff
@@ -846,6 +856,31 @@ def process_arrays(self, context):
 
         return arrays_to_create, array_children
 
+    @staticmethod
+    def recreate_assembly(old_to_new):
```
Comment (brunoperdigao) [inline src/blenderbim/blenderbim/bim/module/geometry/operator.py:860]: Should this be moved to `.../tool/root.py` ?

**Comment by brunoperdigao** [comment]
@Moult I noticed that the BIM tests only account for the Blender side of things. I want to create a test to check wall connections after duplicating them, but I'm not finding a way to this using only Blender, without using IfcOpenShell. Are connections stored in Blender object properties?

**Comment by theoryshaw** [comment]
@brunoperdigao this this pr, have this functionality? 

image

**Comment by brunoperdigao** [comment]
No, this PR only changes things in the normal duplication (shift + d). I need this to be merged first in order to create the PR for the duplicate aggregate refactor. But once this is ready it should be easy to implement the rest.
For this PR it will be useful to test 3 things:
- Select all the objects of an aggregate and shift + d, it should duplicate it (but without any linking property)
- Select two objects that share connection and duplicate them. The connection in the new objects should be reconstructed like the original ones
- The same as the second but with walls that share a connection

**Comment by theoryshaw** [comment]
Thanks!  I assuming it was this branch, correct? https://github.com/IfcOpenShell/IfcOpenShell/compare/v0.7.0...brunoperdigao:IfcOpenShell:refactor-duplicate-aggregate

**Comment by brunoperdigao** [comment]
Yes, I think I'll create the PR already as a draft so you can give me some insights

**Comment by Moult** [comment]
@brunoperdigao there's nothing wrong with checking IOS IFC data in the BIM tests, just that we haven't done so yet (simply because either it's already captured elsewhere or the test was only written as a smoke test not a comprehensive test). Go ahead and write new sentence definitions for it :)

**Comment by theoryshaw** [comment]
> Yes, I think I'll create the PR already as a draft so you can give me some insights

sounds great. Have some ideas. :)

**Comment by brunoperdigao** [comment]
@Moult I finished all the changes that I had planned for this. Let me know if you find there is anything else to improve.

**Pull request closed without merging**